enzy_htp/structure/structure_region/structure_region.py

```diff
@@ -311,6 +318,65 @@ def topology(self) -> Structure:
         for atom in self.atoms:
             if not atom.parent.is_residue_cap():
                 return atom.root()
+            
+    def convert_to_structure(self, cap_as_residue = False) -> Structure:
+        """Converts all atoms in the region to a Structure
+        Args:
+            cap_as_residue: will not include the cap in the residue if True
+        Returns:
+            The corresponding Structure of the structure region"""
+        residue_mapper: {Residue, List[Atom]} = {}
+        res_to_cap_atoms: {Residue, int} = {}
+
+        for aa in self.atoms:
+            if isinstance(aa.parent, ResidueCap):
+                if cap_as_residue:
+                    continue
+                else:
+                    aa.residue = aa.parent.link_atom.residue
+                if aa.residue not in res_to_cap_atoms:
+                    res_to_cap_atoms[aa.residue] = 0
+                res_to_cap_atoms[aa.residue] += 1
+
+            if aa.residue not in residue_mapper:
+                residue_mapper[aa.residue] = []
+            residue_mapper[aa.residue].append(aa)
+
+        residues = []
+        for res, atoms in residue_mapper.items():
+            if res not in res_to_cap_atoms.keys():
+                res_to_cap_atoms[res] = 0
+            
+            if res.rtype == chem.ResidueType.MODIFIED:
+                new_res = ModifiedResidue(int(res.idx), res.name, atoms, res.parent, net_charge=int(res.net_charge), multiplicity=int(res.multiplicity))
+                new_res.renumber_atoms(range(1, len(atoms) + res_to_cap_atoms[res] + 1))
```

Comment:
What is the reason for using res_to_cap_atoms? I think len(atoms) already included the cap atoms right?

Reply:
You're completely right, after adding the new involved_residue_atom_mapper function I was able to remove this other Dict

Reply:
However, it still seems I have to renumber them here due to an error in sorting when I attempt to simply renumber the structure at the end. This is because of the resiude cap atoms not being indexed yet